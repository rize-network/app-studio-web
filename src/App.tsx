import React, { Suspense } from 'react';
import { Route, Routes, useLocation } from 'react-router-dom';
import { ComponentRouter } from 'src/routes/routes';
import { View, Text } from 'app-studio';

export const App: React.FC = () => {
  const location = useLocation();
  return (
    <Suspense
      fallback={
        <View
          height="100vh"
          display="flex"
          alignItems="center"
          justifyContent="center"
        >
          <Text>Loading...</Text>
        </View>
      }
    >
      <Routes location={location}>
        {ComponentRouter.map((item, index) => (
          <Route key={index} path={item.path} element={item.element} />
        ))}
      </Routes>
    </Suspense>
  );
};

export default App;
