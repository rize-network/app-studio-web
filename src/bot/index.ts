import { Bot } from './Bot';
import { DocuCode } from './DocuCode';

async function main() {
  const projectName = 'MDX Documentation';
  const descriptionPath = 'docs/description.md';

  // Get componentName from command line arguments
  const componentName = process.argv[2]; // Assuming it's the first argument passed
  if (!componentName) {
    console.error('Please provide a component name as an argument.');
    process.exit(1);
  }

  console.log(`Generating documentation for component: ${componentName}`);

  const basePath = 'src/components';
  const propsPath = `src/data/props/${componentName}.props.json`;
  const componentFolder = `${basePath}/${componentName}`;
  const componentPath = `${basePath}/${componentName}/${componentName}`;

  const assistantGPT = new Bot(componentName);

  // Implementation for Documentation of the code
  console.log('Initializing documentation process...');
  const docuCode = new DocuCode(componentName);
  const assistantDocumentation = await assistantGPT.init(
    projectName,
    [],
    descriptionPath,
    'docu'
  );

  await docuCode.processDirectory(componentFolder, assistantDocumentation.id);

  // Generating props file for code
  console.log('Generating props file...');
  const { fileIds } = await assistantGPT.addFiles(componentPath);

  const assistantCreation = await assistantGPT.init(
    projectName,
    fileIds,
    descriptionPath,
    'props'
  );

  await assistantGPT.response(assistantCreation.id, propsPath);

  await assistantGPT.MarkdownGeneration(
    componentFolder,
    componentName,
    propsPath
  );

  console.log('Documentation generation completed.');
}

main().catch((error) => {
  console.error('Error during the execution of main:', error);
});
