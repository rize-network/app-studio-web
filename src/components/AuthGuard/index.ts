export { AuthGuard, withAuthGuard, RouteAuthGuard, AuthStatus } from './AuthGuard';
