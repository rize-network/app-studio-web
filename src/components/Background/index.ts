export { Background } from './Background';
export type {
  BackgroundProps,
  AuroraBackgroundProps,
  MeteorsProps,
  WallProps,
  ParticlesProps,
  GridProps,
  RipplesProps,
  BackgroundImageProps,
  BackgroundVideoProps,
  BackgroundGradientProps,
  BackgroundStyles,
} from './Background/Background.props';
export type {
  BackgroundEffect,
  AuroraDirection,
  MeteorTiming,
  BorderAnimationStyles,
  StrokeAnimationStyles,
  BackgroundContextType,
  BackgroundSize,
  BackgroundPosition,
  BackgroundRepeat,
  BlendMode,
} from './Background/Background.type';
