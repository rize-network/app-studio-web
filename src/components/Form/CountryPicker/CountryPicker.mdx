### **Import**

```jsx static
import { CountryPicker } from "@app-studio/web";
```

### **Default**

```jsx
<CountryPicker label="Country" placeholder="Select a country..." />
```

### **Disabled**

“**_isDisabled_**” makes the field unusable.

```tsx
<CountryPicker isDisabled />
```

### **Read Only**

“**_isReadOnly_**” makes the field readable but not writable. Can only read the element.

```tsx
<CountryPicker isReadOnly />
```

### **Size**

“**_size_**” changes the text and icon size . It has a default value of “md”.

```tsx
import { Vertical } from "../../Layout/Vertical/Vertical";

<Vertical gap={10}>
  <CountryPicker name="xs" size="xs" label="xs" />
  <CountryPicker name="sm" size="sm" label="sm" />
  <CountryPicker name="md" size="md" label="md" />
  <CountryPicker name="lg" size="lg" label="lg" />
  <CountryPicker name="xl" size="xl" label="xl" />
</Vertical>;
```

### **Shape**

“**_shape_**” changes the appearance of the borders.

```tsx
import { Vertical } from "../../Layout/Vertical/Vertical";

<Vertical gap={10}>
  {["default", "sharp", "rounded", "pillShaped"].map((shape) => (
    <CountryPicker
      key={shape}
      shape={shape}
      placeholder="Select a country..."
    />
  ))}
</Vertical>;
```

### **Variant**

“**_variant_**” modifies the layout or design of the component.

```tsx
import { Vertical } from "../../Layout/Vertical/Vertical";

<Vertical gap={10}>
  {["outline", "default", "none"].map((variant) => (
    <CountryPicker
      key={variant}
      variant={variant}
      label={variant}
      placeholder="Select a country..."
    />
  ))}
</Vertical>;
```

### **ColorScheme**

“**_colorScheme_**” modifies the label and border color of the field.

```jsx
import { Vertical } from "../../Layout/Vertical/Vertical";

<Vertical gap={15}>
  {["theme.primary", "theme.secondary", "theme.error", "theme.success", "theme.warning"].map((color) => (
    <CountryPicker key={color} label={color} colorScheme={color} />
  ))}
</Vertical>;
```

### **Shadow**

“**_shadow_**” adds a shadow effect to the field.

```tsx
<CountryPicker shadow={{ boxShadow: "rgba(0, 0, 0, 0.20) 0px 3px 8px" }} />
```

### **HelperText**

“**_helperText_**” provides additional information about the field.

```tsx
<CountryPicker helperText="CountryPicker one item!" error />
```

### **Styles**

“**_styles_**” changes the style of the field.

```jsx
<CountryPicker
  styles={{
    box: { 
      borderRadius: 10, 
      borderColor: 'theme.primary',
      borderStyle: 'solid',
      borderWidth: 1
    },
    text: { color: "theme.primary"  },
    label: { fontWeight: "bold", color: "theme.primary"  },
    dropDown: { color: "theme.primary"  },
  }}
  error
/>
```

```jsx static
const Shapes: Record<Shape, number | string> = {
  default: "6px 6px 0 0",
  sharp: 0,
  rounded: 4,
  pillShaped: 24,
};
```

## Types

```tsx static
type Size = "xs" | "sm" | "md" | "lg" | "xl";
```

```jsx static
type CountryPickerStyles = {
  text?: CSSProperties,
  icon?: CSSProperties,
  dropDown?: CSSProperties,
  box?: CSSProperties,
  label?: CSSProperties,
  helperText?: CSSProperties,
};
```

```jsx static
type Shape = "default" | "sharp" | "rounded" | "pillShaped";
```

```jsx static
type Country = {
  name: string,
  dial_code: string,
  emoji: string,
  code: string,
};
```

```js static
type Variant = "default" | "outline" | "none";
```

```jsx static
type Styles = {
  box?: CSSProperties,
  text?: CSSProperties,
  label?: CSSProperties,
  helperText?: CSSProperties,
  field?: CSSProperties,ssss
};
```

## Default Values

```jsx static
const LabelSizes: Record<Size, number> = {
  xs: 8,
  sm: 10,
  md: 12,
  lg: 14,
  xl: 16,
};
```

```jsx static
const IconSizes: Record<Size, string> = {
  xs: "8px",
  sm: "10px",
  md: "12px",
  lg: "14px",
  xl: "16px",
};
```
