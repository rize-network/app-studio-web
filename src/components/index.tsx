export * from './Button/Button';
export * from './Form/Checkbox/Checkbox';
export * from './Form/CountryPicker/CountryPicker';
export * from './Form/DatePicker/DatePicker';
export * from './Form/Password/Password';
export * from './Form/Select/Select';
export * from './Form/Switch/Switch';
export * from './Form/Switch/Switch';
export * from './Form/TextArea/TextArea';
export * from './Form/TextField/TextField';
export * from './Layout/Center/Center';
export * from './Formik';
export * from './Layout/Horizontal/Horizontal';
export * from './Layout/Vertical/Vertical';
export * from './Layout/View/View';
export * from './Link/Link';
export * from './Loader/Loader';
export * from './Modal/Modal';
export * from './Text/Text';
export * from './File/File';
export * from './Tabs/Tabs';
export * from './Alert/Alert';
export * from './AspectRatio/AspectRatio';
export * from './Avatar/Avatar';
export * from './Badge/Badge';
export * from './Toggle/Toggle';
export * from './ToggleGroup/ToggleGroup';
export * from './Message/Message';
