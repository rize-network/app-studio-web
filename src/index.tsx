import React from 'react';
import ReactDOM from 'react-dom/client';
import AppProvider from 'src/providers';
import { App } from './App';
import { ComboBoxProvider } from './components/Form/ComboBox/ComboBox/ComboBox.provider';

const root = ReactDOM.createRoot(
  document.getElementById('root') as HTMLElement
);
root.render(
  <React.StrictMode>
    <AppProvider>
      <ComboBoxProvider>
        <App />
      </ComboBoxProvider>
    </AppProvider>
  </React.StrictMode>
);
