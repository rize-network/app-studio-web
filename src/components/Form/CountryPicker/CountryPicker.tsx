import React from 'react';
import { CountryPickerProps } from './CountryPicker/CountryPicker.props';
// Uses the custom hook 'useCountryPickerState' to manage state specific to the CountryPicker component.
import { useCountryPickerState } from './CountryPicker/CountryPicker.state';
import CountryPickerView from './CountryPicker/CountryPicker.view';
// Defines the CountryPickerComponent as a functional component with type 'React.FC' that takes 'CountryPickerProps' as props.
const CountryPickerComponent: React.FC<CountryPickerProps> = (props) => {
  // Invokes the 'useCountryPickerState' hook to obtain state and methods related to country picking functionality.
  const countryPickerStates = useCountryPickerState(props);
  // Renders the CountryPickerView component passing in state and props for dynamic UI representation.
  return <CountryPickerView {...countryPickerStates} {...props} />;
};
// Exports the CountryPickerComponent for use in other parts of the application as 'CountryPicker'.
export const CountryPicker = CountryPickerComponent;
