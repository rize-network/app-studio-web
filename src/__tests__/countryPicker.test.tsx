import React from 'react';
import renderer from 'react-test-renderer';
import { CountryPicker } from 'src/components';
import countries from 'src/components/Form/CountryPicker/countries.json';
import { DropDown, DropDownItem } from 'src/components/Form/CountryPicker/CountryPicker/CountryPicker.view';

import { cleanup, fireEvent, render, screen } from '@testing-library/react';

afterEach(() => {
  cleanup();
});

describe('CountryPicker component', () => {
  test('renders CountryPicker component', () => {
    render(<CountryPicker name="username" role="textbox" />);
    const CountryPickerElement = screen.getByRole('textbox');
    expect(CountryPickerElement).toBeInTheDocument();
  });
  test('renders CountryPicker with specified role', () => {
    render(<CountryPicker name="username" role="textbox" />);
    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toBeInTheDocument();
  });
  test('renders with placeholder', () => {
    render(<CountryPicker name="username" placeholder="Enter your name" />);
    const inputElement = screen.getByPlaceholderText('Enter your name');
    expect(inputElement).toBeInTheDocument();
  });
  test('renders disabled CountryPicker', () => {
    render(<CountryPicker name="username" isDisabled />);
    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toBeDisabled();
  });
  test('triggers onBlur event when the input field loses focus', () => {
    const handleBlur = jest.fn();
    render(<CountryPicker onBlur={handleBlur} />);
    const countryInput = screen.getByRole('textbox');
    fireEvent.focus(countryInput);
    fireEvent.blur(countryInput);
    // Assert that the onBlur event is triggered
    expect(handleBlur).toHaveBeenCalled();
  });
  test('triggers onChange event when selecting a country', () => {
    const handleChange = jest.fn();
    render(<CountryPicker onChange={handleChange} />);
    const countryInput = screen.getByRole('textbox');
    const selectedCountry = 'United States';
    fireEvent.change(countryInput, { target: { value: selectedCountry } });
    // Assert that the onChange event is triggered with the selected country value
    expect(handleChange).toHaveBeenCalledWith(selectedCountry);
  });
  test('renders readonly CountryPicker', () => {
    render(<CountryPicker name="username" value="readonly value" isReadOnly />);
    const inputElement = screen.getByRole('textbox');
    expect(inputElement).toHaveAttribute('readonly');
  });
  test('renders helper text', () => {
    const helperText = 'This is helper text';
    render(<CountryPicker name="error" helperText={helperText} error={true} />);
    const helperTextElement = screen.queryByText(helperText);
    expect(helperTextElement).toBeInTheDocument();
  });
  test('CountryPicker to match snapshot', () => {
    const tree = renderer
      .create(
        <CountryPicker
          size="xl"
          name="name"
          error={true}
          placeholder="Name"
          helperText={'required'}
          isAutoFocus
          shape="pillShaped"
          variant="outline"
          styles={{
            text: { color: 'theme.primary' },
            icon: { width: 0.2 },
            label: { fontSize: 15 },
            dropDown: { backgrounColor: 'theme.primary' },
            helperText: { color: 'red' },
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('DropDown component', () => {
  const options = countries;
  const callback = jest.fn();
  const styles = {
    dropDown: { backgroundColor: 'red' },
    text: { color: 'white' },
  };

  test('renders DropDown component without crashing', () => {
    render(<DropDown options={options} styles={styles} />);
    const DropDownElement = screen.getByRole('dropDown');
    expect(DropDownElement).toBeInTheDocument();
  });

  test('applies custom styles to the dropdown list', () => {
    render(<DropDown options={options} callback={callback} styles={styles} />);
    const dropDownList = screen.getByRole('dropDown');
    expect(dropDownList).toHaveStyle('background-color: red');
  });

  test('renders options correctly', () => {
    const options = [
      {
        name: 'Aland Islands',
        dial_code: '+358',
        emoji: 'ðŸ‡¦ðŸ‡½',
        code: 'AX',
      },
      {
        name: 'Albania',
        dial_code: '+355',
        emoji: 'ðŸ‡¦ðŸ‡±',
        code: 'AL',
      },
    ];
    render(<DropDown options={options} callback={() => {}} styles={styles} />);

    options.forEach((option) => {
      const optionElement = screen.getByText(option.name);
      expect(optionElement).toBeInTheDocument();
    });
  });

  test('renders DropDown component with provided size', () => {
    render(<DropDown options={options} styles={styles} size="md" />);
    const DropDownElement = screen.getByRole('dropDown');
    expect(DropDownElement).toBeInTheDocument();
    expect(DropDownElement).toHaveStyle({ fontSize: 16 });
  });

  test('calls callback function with selected option', () => {
    const options = [
      {
        name: 'Aland Islands',
        dial_code: '+358',
        emoji: 'ðŸ‡¦ðŸ‡½',
        code: 'AX',
      },
      {
        name: 'Albania',
        dial_code: '+355',
        emoji: 'ðŸ‡¦ðŸ‡±',
        code: 'AL',
      },
    ];
    const callback = jest.fn();
    render(<DropDown options={options} callback={callback} styles={styles} />);

    const selectedOption = options[1];
    const optionElement = screen.getByText(selectedOption.name);
    fireEvent.click(optionElement);

    expect(callback).toHaveBeenCalledWith(selectedOption.name);
  });
});

describe('DropDownItem component', () => {
  const option = 'Albania';
  const styles = {
    text: { color: 'red' },
  };
  test('renders DropDownItem component without crashing', () => {
    render(<DropDownItem option={option} callback={() => {}} styles={styles} />);
    const DropDownItemElement = screen.getByRole('DropDownItem');
    expect(DropDownItemElement).toBeInTheDocument();
  });

  test('calls callback function when option is clicked', () => {
    const option = 'United States';
    const callback = jest.fn();
    render(<DropDownItem option={option} callback={callback} styles={styles} />);

    const optionElement = screen.getByText(option);
    fireEvent.click(optionElement);

    expect(callback).toHaveBeenCalledWith(option);
  });

  test('changes background color on hover', () => {
    const option = 'United States';
    render(<DropDownItem option={option} callback={() => {}} styles={styles} />);

    const optionElement = screen.getByText(option);
    fireEvent.mouseEnter(optionElement);

    expect(optionElement).toHaveStyle('background-color: trueGray.100');

    fireEvent.mouseLeave(optionElement);

    expect(optionElement).not.toHaveStyle('background-color: trueGray.100');
  });

  test('renders DropDownItem component with provided size', () => {
    render(<DropDownItem option={option} selected="Option 1" size="md" callback={() => {}} styles={styles} />);
    const DropDownItemElement = screen.getByRole('DropDownItem');
    expect(DropDownItemElement).toBeInTheDocument();
    expect(DropDownItemElement).toHaveStyle({ fontSize: 16 });
  });

  test('should call callback when an option is selected', () => {
    const callback = jest.fn();
    render(<DropDownItem option={option} selected="Option 2" callback={callback} styles={styles} />);
    fireEvent.click(screen.getByRole('DropDownItem'));
    expect(callback).toHaveBeenCalledTimes(1);
  });
});
