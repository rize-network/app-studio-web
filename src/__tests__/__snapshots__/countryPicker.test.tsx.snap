// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryPicker component CountryPicker to match snapshot 1`] = `
<div
  className="sc-bdfCDU bChUJt"
  display="flex"
  flexDirection="column"
  flexWrap="nowrap"
  gap={5}
  justifyContent="flex-start"
  onClick={[Function]}
  position="relative"
>
  <div
    alignItems="center"
    backgroundColor="trueGray.50"
    borderColor="error"
    borderRadius={24}
    borderStyle="solid"
    borderWidth={1}
    className="sc-bdfCDU drLOlu"
    color="theme.primary"
    cursor="pointer"
    display="flex"
    filter="brightness(1)"
    flexDirection="row"
    flexWrap="nowrap"
    fontSize={20}
    gap={10}
    justifyContent="space-between"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    paddingLeft={16}
    paddingRight={35}
    paddingVertical={8}
    transition="padding 0.2s ease"
    width="100%"
  >
    <div
      className="sc-bdfCDU jCBzwc"
      display="flex"
      flexDirection="column"
      flexWrap="nowrap"
      justifyContent="flex-start"
      width="100%"
    >
      <input
        autoFocus={true}
        backgroundColor="transparent"
        border="none"
        className="sc-bdfCDU cbFLIy"
        color="color.blueGray.700"
        cursor="auto"
        disabled={false}
        fontSize={20}
        heigth="100%"
        margin={0}
        name="name"
        on={
          Object {
            "focus": Object {
              "outline": "none",
            },
          }
        }
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        paddingHorizonatl={0}
        paddingVerical={8}
        placeholder="Name"
        readOnly={false}
        type="country"
        value="Name"
        width="100%"
      />
    </div>
    <div
      alignItems="center"
      className="sc-bdfCDU gYnDEY"
      display="flex"
      flexWrap="nowrap"
      gap={10}
      justifyContent="center"
      position="absolute"
      right={16}
      top="50%"
      transform="translateY(-50%)"
      zIndex={500}
    >
      <div
        alignItems="center"
        className="sc-bdfCDU hcIsQh"
        display="flex"
        height="16px"
        justifyContent="center"
        width="16px"
      >
        <svg
          fill="#000000"
          style={
            Object {
              "width": 0.2,
            }
          }
          version="1.1"
          viewBox="0 -4.5 20 20"
        >
          <g
            id="SVGRepo_bgCarrier"
            strokeWidth="0"
          />
          <g
            id="SVGRepo_tracerCarrier"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
          <g
            id="SVGRepo_iconCarrier"
          >
            <title>
              arrow_down [#338]
            </title>
             
            <desc>
              Created with Sketch.
            </desc>
            <defs>
               
            </defs>
            <g
              fill="none"
              fillRule="evenodd"
              id="Page-1"
              stroke="none"
              strokeWidth="1"
            >
              <g
                fill="#000000"
                id="Dribbble-Light-Preview"
                transform="translate(-220.000000, -6684.000000)"
              >
                <g
                  id="icons"
                  transform="translate(56.000000, 160.000000)"
                >
                  <path
                    d="M164.292308,6524.36583 L164.292308,6524.36583 C163.902564,6524.77071 163.902564,6525.42619 164.292308,6525.83004 L172.555873,6534.39267 C173.33636,6535.20244 174.602528,6535.20244 175.383014,6534.39267 L183.70754,6525.76791 C184.093286,6525.36716 184.098283,6524.71997 183.717533,6524.31405 C183.328789,6523.89985 182.68821,6523.89467 182.29347,6524.30266 L174.676479,6532.19636 C174.285736,6532.60124 173.653152,6532.60124 173.262409,6532.19636 L165.705379,6524.36583 C165.315635,6523.96094 164.683051,6523.96094 164.292308,6524.36583"
                    id="arrow_down-[#338]"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
  <div
    className="sc-bdfCDU VdVeV"
    color="theme.error"
    dropDown={
      Object {
        "backgrounColor": "theme.primary",
      }
    }
    fontSize={12}
    fontStyle="normal"
    fontWeight={400}
    helperText={
      Object {
        "color": "red",
      }
    }
    icon={
      Object {
        "width": 0.2,
      }
    }
    label={
      Object {
        "fontSize": 15,
      }
    }
    lineHeight={12}
    marginHorizontal={0}
    marginVertical={0}
    role="text"
    text={
      Object {
        "color": "theme.primary",
      }
    }
    textDecoration="none"
  >
    required
  </div>
</div>
`;
